rpc/src/rpc.rs

```diff
@@ -2338,13 +2324,15 @@ fn get_spl_token_owner_filter(program_id: &Pubkey, filters: &[RpcFilterType]) ->
     for filter in filters {
         match filter {
             RpcFilterType::DataSize(size) => data_size_filter = Some(*size),
+            #[allow(deprecated)]
```

Comment:
I was feeling really good about this PR, until these annotations. I haven't yet thought of a clever alternative that allows this level of matching complexity.
Maybe the answer is to be less clever, because these are internal methods and we should have already run Memcmp::convert_to_raw_bytes()...?

Reply:
Yeah, I was initially thinking helpers, but I haven't yet come up with an elegant way to write it yet, especially as there's already a `bytes()` method. Maybe I'll sleep on it.

Reply:
Everything I've tried feels worse, so I think I'm happy with the annotations for now.

Reply:
fine by me!

least dumb thing i can think of is to move the match guard logic to a method on `Memcmp` that returns `Option<NewThing>`, where...
```rust
enum NewThing {
    Filter(Vec<u8>),
    TokenOwnerFull(Pubkey),
    TokenOwnerPartial(usize),
}
```
`match` on that here and fill out the locals as appropriate.